bookkeeper-server/src/main/java/org/apache/bookkeeper/client/api/BookKeeperException.java

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright 2017 The Apache Software Foundation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.bookkeeper.client.api;
+
+import org.apache.bookkeeper.client.LedgerHandleAdv;
+
+/**
+ * Super class for all errors which occur using BookKeeper client
+ */
+public abstract class BookKeeperException extends Exception {
```

Comment:
can we stick to 'BKException'? otherwise, when we deprecate the old class, it will cause a lot of renames. ideally it should be changing imports.